### PR TITLE
Improve text reporter

### PR DESCRIPTION
**Pull request by ybiquitous**
This change aims to make the `--format=text` output more useful and helpful.

- Add summary about files and issues
- Change the output format like the [Clang style](https://clang.llvm.org/diagnostics.html).

For example:

```console
$ exe/goodcheck check README.md
README.md:7:29: Did you mean "GitHub"?
What if a misspelling like `Github` for `GitHub` can be found automatically?
                            ^~~~~~

1 file inspected, 1 issue detected
```

image

### CODE REVIEW

File: test/check_command_test.rb [hunk 1]
```diff
@@ -327,9 +389,16 @@ def test_encoding
         check = Check.new(config_path: builder.config_path, rules: [], targets: [Pathname(".")], reporter: reporter, stderr: stderr, force_download: false, home_path: builder.path + "home")
 
         assert_equal 2, check.run
+        assert_equal <<OUT, stdout.string
+euc_jp.txt:1:10: Foo
+吾輩は猫である。
+         ^~~
```
Comment (yykamei) [inline test/check_command_test.rb:395]: [nitpick] The byte size of "吾輩は" encoded with EUC-JP should be `6`; 2 bytes x 3 items.

By the way, I think a multibyte character can be assumed to have the width of `2` in many cases even though it consists of three or four bytes.

Comment (ybiquitous) [reply test/check_command_test.rb:395]: Assuming the multibyte character as 2 widths might be useful, but the current implementation is based on bytes.
So, we may need to change more code and be more complex.

Comment (yykamei) [reply test/check_command_test.rb:395]: Thank you. I'm happy that Goodcheck appropriately displays the mark for multibyte characters.

**Review by yykamei — APPROVED** [review]
LGTM

**Pull request merged**